### ISSUE TITLE
Upgrade Ubuntu version in CI images

### ISSUE THREAD

**Issue by spring-buildmaster** [spring-projects/spring-boot#20679]
Upgrade to ubuntu:bionic-20200311

**Comment by dreis2211** [comment]
I wonder why there was a second issue created. As I can't take a look at the logs of https://ci.spring.io/teams/spring-boot/pipelines/spring-boot-2.3.x/jobs/detect-ubuntu-image-updates/builds/5 I can't tell why.

**Comment by snicoll** [comment]
Believe it or not, I can't either (on that particular build)

**Comment by snicoll** [comment]
With the help of @mbhave and @wilkinsona, the reason why I can't expand that particular build is because it didn't log anything.

**Comment by snicoll** [comment]
race condition?

**Comment by dreis2211** [comment]
Well, it still needs to be executed twice somehow to create two tasks that could race against each other. Anyhow, the logic is the same as in the JDK updates so it theoretically suffers from the same problem.
Should we close this one and open a new issue to investigate?

**Comment by wilkinsona** [comment]
Yep. It's a distributed race condition. The check ran in the 2.1.x, 2.2.x, and 2.3.x pipelines. 2.1.x was a bit slow and detected that the issue already existed. 2.2.x and 2.3.x both checked in parallel, didn't find the existing issue, and then both created one.

**Comment by dreis2211** [comment]
Ah, between the different branches. Of course.
Should we shuffle the timings a bit in the pipeline.yml?

**Comment by wilkinsona** [comment]
I'm not sure if we should do that, or check for and create branch-specific issues.

**Comment by mbhave** [comment]
I think we should create branch specific issues and assign the milestone at the time of issue creation.

**Comment by dreis2211** [comment]
It might be a bit difficult to assign 2.2.6 or 2.3.0.M4. 2.2.x or 2.3.x should be relatively simple, though. Which means someone needs to assign the specific milestone manually later, I guess (if I understand the release notes creation correctly). What do you think?

**Comment by mbhave** [comment]
That's consistent with other issues where we manually change the milestone for closed issues from the `.x` version to the specific version.

**Comment by dreis2211** [comment]
Mind if I open a new issue to track these efforts?

**Comment by mbhave** [comment]
Actually, I'm having second thoughts about this. If we create the same issue in multiple milestones we need to remember to not create the `forward-port` issue when forward-merging. Instead the forward-merge needs to close the issue that has already been created by the automation. I don't think it's a big deal but it does break the workflow a bit.

**Comment by mbhave** [comment]
I'll close this one in favor of #20680 in any case. @dreis2211 let's continue the discussion on that one.
